common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -345,22 +360,29 @@ def is_user_authenticated(self, context):
         # NOTE (CCB): We default to true to maintain the behavior in place prior to allowing anonymous access access.
         return context.get('user_authenticated', True)
 
-    def _student_or_public_view(self, context, prereq_met, prereq_meta_info, banner_text=None, view=STUDENT_VIEW):
-        """
-        Returns the rendered student view of the content of this
-        sequential.  If banner_text is given, it is added to the
-        content.
-        """
-        _ = self.runtime.service(self, "i18n").ugettext
+    def student_view_data(self, context=None):
+        context = context or {}
+        context = {'exclude_units': True}
+        prereq_met = True
+        prereq_meta_info = {}
+        banner_text = None
         display_items = self.get_display_items()
-        self._update_position(context, len(display_items))
 
+        if self._required_prereq():
+            if self.runtime.user_is_staff:
```

Comment:
The `student_view_data` method [is called](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/course_api/blocks/transformers/student_view.py#L62-L63) during the `collect` phase of the `StudentViewTransformer`. Because that phase happens right at the time of Studio publish, there is no student state available for it to bind to at the time, and we can't make any operations that rely on reading anything user-specific. You can put this identical functionality in a new method though.

Reply:
It turns out, this method isn't needed. I was hoping to include the sequence metadata in the block API response, which would require implementing `student_view_data`, but since we're going to make a separate request for the sequence (to `xmodule_handler/metadata`), this one can be removed.